src/apps/search-result/helper.ts

```diff
@@ -18,4 +18,13 @@ export const getFirstMaterialTypeFromFilters = (
     : undefined;
 };
 
+export const shouldShowInvalidSearchPage = (
+  q: string,
+  qUrlParameter: string | null
+) => {
+  return (
+    (!qUrlParameter && !q) || (!qUrlParameter && q === "null") || q.length < 3
```

Comment:
I think it is a flaw that `q` can be "null" it should not be possible.
These changes to `getParams` should fix that:
```typescript
export const getParams = (props: Record<string, string | undefined>) =>
  Object.entries(props).reduce<Record<string, string>>(
    (acc, [property, value]) => {
      const paramValue = value || getUrlQueryParam(property);
      return {
        ...acc,
        [property]: paramValue ? String(paramValue) : ""
      };
    },
    {}
  );

Reply:
Also it should not be necessary to check for `q.length` if we add this to the search query:
```typescript
  const { data, isLoading } = useSearchWithPaginationQuery(
    {
      q: { all: q },
      offset: page * pageSize,
      limit: pageSize,
      filters: createFilters(filters, cleanBranches)
    },
    {
      enabled: !isEmpty(q) && q.length > 2
    }
  );

Reply:
... so I think it is nice to have a helper function called: `shouldShowInvalidSearchPage`. But the logic of it should be revised after my proposals :)

Reply:
Very good suggestion with the revision of `getParams` helper, I added it in. But I don't necessarily agree with your second point, and I think that it's a matter of preference. 

If we don't ever make a call and we already are on the search result page, then we're never gonna get any data, which currently just returns `null` instead of the whole component. We could of course then return our new `< SearchResultInvalidSearch  />` component, but are we sure that that's always the correct case there? Asking a dedicated function whether to render the `< SearchResultInvalidSearch  />` component based on all the criteria that we need to know about in order to make the right call seems like the better options to me.

Reply:
It didn't really make sense to keep it - the statement is tiny now :)

Great suggestion with refactoring the `getParams ` helper